### PR TITLE
[IOS-526] I don't see flash message when I am on the bottom of the page

### PR DESCRIPTION
**Pull request by bwdowikowski**
### Ticket
[IOS-526](https://netguru.atlassian.net/browse/IOS-526)

### Task Description
Added possibility that flashmessage will not be displayed in current view but in keyWindow instead

### CODE REVIEW

File: Inbbbox/Source Files/ViewControllers/SettingsViewController.swift [hunk 1]
```diff
@@ -65,16 +65,21 @@ extension SettingsViewController: ModelUpdatable {
...
         tabBarController?.present(alert, animated: true, completion: nil)
     }
 }
 
+// MARK: FlashMessageDisplayable
+
 extension SettingsViewController: FlashMessageDisplayable {
     func displayFlashMessage(_ model: FlashMessageViewModel) {
-        FlashMessage.sharedInstance.showNotification(inViewController: self, title: model.title, canBeDismissedByUser: true)
+//        FlashMessage.sharedInstance.showNotification(inViewController: self, title: model.title, canBeDismissedByUser: true)
```
Comment (pikor) [inline Inbbbox/Source Files/ViewControllers/SettingsViewController.swift]: Not needed.

File: Inbbbox/Source Files/Views/Custom/FlashMessageView.swift [hunk 2]
```diff
@@ -167,16 +176,16 @@ final class FlashMessageView: UIView {
     }
     
     fileprivate func setupPosition() {
-        guard let viewController = viewController else {
-            return
-        }
+//        guard let viewController = viewController else {
```
Comment (pikor) [inline Inbbbox/Source Files/Views/Custom/FlashMessageView.swift]: Don't you want to guard for this case?

Comment (bwdowikowski) [inline Inbbbox/Source Files/Views/Custom/FlashMessageView.swift]: No it's not needed, as you can see I did take care about it. Like in lines `183`, `188`. So it's covered.

**Comment by pikor** [comment]
Also, out of curiosity, to show flash message, this class uses 
`flashMessageView.displayInView?.insertSubview(flashMessageView, belowSubview: currentNavigationController.navigationBar)`
Did you try to show it by adding it to current window?

**Comment by bwdowikowski** [comment]
Yea, this would work in this case. But what with situations when you'll have navigationController without scroll then message view will be inserted behind navigation bar. 
On second thought it would be good to be consistent about the way of displaying regardless of if we have navigation bar or not.  Yea `addSubview` is better solution here.

**Comment by netguru-hound** [comment]
## [Current coverage](https://codecov.io/gh/netguru/inbbbox-ios/pull/361?src=pr) is 31.97% (diff: 56.52%)
> Merging [#361](https://codecov.io/gh/netguru/inbbbox-ios/pull/361?src=pr) into [develop](https://codecov.io/gh/netguru/inbbbox-ios/branch/develop?src=pr) will increase coverage by **0.04%**

```diff
@@            develop       #361   diff @@
==========================================
  Files           260        260          
  Lines         11798      11795     -3   
  Methods           0          0          
  Messages          0          0          
  Branches          0          0          
==========================================
+ Hits           3767       3771     +4   
+ Misses         8031       8024     -7   
  Partials          0          0          
```

> Powered by [Codecov](https://codecov.io?src=pr). Last update [c3262b4...57eeb24](https://codecov.io/gh/netguru/inbbbox-ios/compare/c3262b4d2de07ec18320d98b5e24de1f312e7cc8...57eeb24b9cd372facee978cea28747da6d2275d9?src=pr)

**Pull request merged**